src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java

```diff
@@ -778,6 +778,18 @@ public void scrollToPosition(String selector) {
     jsActions.scrollToSpecificElement(driver.findElement(By.cssSelector(selector)));
     PageObjectLogging.log("scrollToSelector", "Scroll to the web selector " + selector, true);
   }
+  // This scroll has been created becouse ad is not displayed if we scroll quickly to the Footer ADEN-4359
+  public void scrollToBottomLeaderboard(){
+    scrollToFooter();
+    scrollTo(presentLeaderboard);
+    scrollToFooter();
+  }
+
+  public void clickOnAdImage(String slotName){
+    wait.forElementVisible(By.cssSelector(AdsContent.getSlotSelector(slotName)));
+    driver.findElement(By.cssSelector(AdsContent.getSlotSelector(slotName))).click();
```

Comment:
you can store `By.cssSelector(AdsContent.getSlotSelector(slotName)` in a variable and reuse it in `wait` and `findElement` methods.